unit_tests/src/test_rc_genquery2.cpp

```diff
@@ -118,4 +118,30 @@ TEST_CASE("rc_genquery2")
             CHECK(result.at(0).at(0).get_ref<const std::string&>() == expected);
         }());
     }
+
+    SECTION("#8182: supports long sequence of whitespace characters between keywords")
+    {
+        // This query string intentionally contains spaces, tabs, and newlines.
+        // The result of the query is the home collection of the user running the test.
+        // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
+        auto query_string = fmt::format(R"(select      		COLL_NAME
+
+
+        		where       	COLL_NAME
+=      				'{}')", env.rodsHome);
+
+        Genquery2Input input{};
+        input.query_string = query_string.data();
+
+        char* output{};
+        REQUIRE(rc_genquery2(static_cast<RcComm*>(conn), &input, &output) == 0);
+        REQUIRE_FALSE(output == nullptr);
```

Comment:
Will `output` leak once we get past this assertion?

Reply:
It sure will.

Nice catch.

Will fix this one and check the other tests too.

Reply:
Fixed.